### PR TITLE
update all urls to be schema-less

### LINKED ISSUE #270: Add HTTPS Support
Kikar.org is currently available only in http, this should change.

### PR DESCRIPTION
**Pull request by yotammanor**
fixes #270.

**Changes Introduced:**
- change all urls to be schema-less (unless handled otherwise)
- add git hook to prevent commit new code that has "http://" in it (or print/set_trace statements)
- add git hook update/installation in vagrantfile.

### CODE REVIEW

**Comment by yotammanor** [comment]
@agamrafaeli do you mind reviewing this change? :)

**Comment by yotammanor** [comment]
@uda is this what you wanted to be done?

File: Vagrantfile [hunk 1]
```diff
@@ -37,6 +37,12 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
     shell.inline = <<-EOS
       set -e
       cd /vagrant/
+      # install/update all git-hooks
+      for hook in $(find git-hooks/ -type f | xargs basename)
```
Comment (aryehpro) [inline Vagrantfile]: Have you run this and seen that it works? I came across this: https://stackoverflow.com/questions/37335560/how-to-extract-file-name-from-piped-results
which says xargs with basename can cause problems (I found  few other references to such problems, and suggestions to use other methods to get the filename itself).

File: git-hooks/pre-commit [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# If you want to allow non-ASCII filenames set this variable to true.
+# allownonascii=$(git config --bool hooks.allownonascii)
+
+# Redirect output to stderr.
+exec 1>&2
+
+DEBUG_FUNC=set_trace
+PRINT_FUNC=print
+NON_SAFE_PROTOCOL="http://"
+for invalid in $DEBUG_FUNC $PRINT_FUNC NON_SAFE_PROTOCOL;
```
Comment (aryehpro) [inline git-hooks/pre-commit]: Why are you invalidating any use of set_trace or print? was this part of a previous git hook (couldn't see it in the source from 'dev' branch)?
I can see why this is a good idea, but it seems unrelated to the rest of the PR, and can be confusing. Maybe add a comment about this \ separate it to a different PR.

File: kikar_hamedina/templates/layouts/old-generic-template.html [hunk 3]
```diff
@@ -7,7 +7,7 @@
 
     <!-- Fonts -->
     <link href="//netdna.bootstrapcdn.com/font-awesome/4.0.3/css/font-awesome.css" rel="stylesheet">
-    <link href='http://fonts.googleapis.com/css?family=Alef:400,700&subset=all' rel='stylesheet' type='text/css'>
+    <link href='//fonts.googleapis.com/css?family=Alef:400,700&subset=all' rel='stylesheet' type='text/css'>
```
Comment (aryehpro) [inline kikar_hamedina/templates/layouts/old-generic-template.html]: Not sure why this is relevant. There doesn't seem to be any usage of this template anywhere.

**Comment by aryehpro** [comment]
Also, 2 places that were missed and I think are relevant for changing:

- kikar-hamedina/kikar_hamedina/mks/models.py (url for OK is with http, this seems to affect al links to OK on MKs pages)
- kikar-hamedina/kikar_hamedina/mks/management/commands/update_mks_url_img.py (this seems like it will affect fetching mks photos

I'm assuming anything under 'reporting/' is not used in the site that is served to users, because there are a few issues in those parts, but if it's for dev use it doesn't relly matter.

File: kikar_hamedina/templates/layouts/old-generic-template.html (hunk 3 above)
Comment (yotammanor) [reply kikar_hamedina/templates/layouts/old-generic-template.html]: removed it.

**Comment by yotammanor** [comment]
@aryehpro belated thanks for the review. I removed the missing things, and responded to your comments. Going to merge it in, so no further action required on your behalf. Cheers!

**Pull request merged**